src/business-logic/services/database-change-ingest-service.ts

```diff
@@ -40,46 +55,170 @@ export class DatabaseChangeIngestService implements IngestChangeService {
-    partChangedListener: DataChangedListener<Part>
+    private readonly ingestedEntityToEntityMapper: IngestedEntityToEntityMapper,
+    rundownChangeListener: DataChangedListener<IngestedRundown>,
+    segmentChangedListener: DataChangedListener<IngestedSegment>,
+    partChangedListener: DataChangedListener<IngestedPart>
   ) {
     this.listenForRundownChanges(rundownChangeListener)
     this.listenForSegmentChanges(segmentChangedListener)
     this.listenForPartChanges(partChangedListener)
+
+    this.enqueueEvent(0, () => this.syncDataWithIngestedData())
+  }
+
+  private listenForRundownChanges(rundownChangeListener: DataChangedListener<IngestedRundown>): void {
+    rundownChangeListener.onCreated(rundown => this.enqueueEvent(1, () => this.createRundown(rundown)))
+    rundownChangeListener.onUpdated(rundown => this.enqueueEvent(6, () => this.updateRundown(rundown)))
+    rundownChangeListener.onDeleted(rundownId => this.enqueueEvent(9, () => this.deleteRundown(rundownId)))
+  }
+
+  private listenForSegmentChanges(segmentChangedListener: DataChangedListener<IngestedSegment>): void {
+    segmentChangedListener.onCreated(segment => this.enqueueEvent(2, () => this.createSegment(segment)))
+    segmentChangedListener.onUpdated(segment => this.enqueueEvent(5, () => this.updateSegment(segment)))
+    segmentChangedListener.onDeleted(segmentId => this.enqueueEvent(8, () => this.deleteSegment(segmentId)))
   }
 
-  private listenForRundownChanges(rundownChangeListener: DataChangedListener<Rundown>): void {
-    rundownChangeListener.onCreated(rundown => this.enqueueEvent(() => this.createRundown(rundown)))
-    rundownChangeListener.onUpdated(rundown => this.enqueueEvent(() => this.updateRundown(rundown)))
-    rundownChangeListener.onDeleted(rundownId => this.enqueueEvent(() => this.deleteRundown(rundownId)))
+  private listenForPartChanges(partChangedListener: DataChangedListener<IngestedPart>): void {
+    partChangedListener.onCreated(part => this.enqueueEvent(3, () => this.createPart(part)))
+    partChangedListener.onUpdated(part => this.enqueueEvent(4, () => this.updatePart(part)))
+    partChangedListener.onDeleted(partId => this.enqueueEvent(7, () => this.deletePart(partId)))
   }
 
-  private listenForSegmentChanges(segmentChangedListener: DataChangedListener<Segment>): void {
-    segmentChangedListener.onCreated(segment => this.enqueueEvent(() => this.createSegment(segment)))
-    segmentChangedListener.onUpdated(segment => this.enqueueEvent(() => this.updatedSegment(segment)))
-    segmentChangedListener.onDeleted(segmentId => this.enqueueEvent(() => this.deleteSegment(segmentId)))
+  private async syncDataWithIngestedData(): Promise<void> {
+    const ingestedRundowns: IngestedRundown[] = await this.ingestedRundownRepository.getIngestedRundowns()
+    await this.deleteRundownsNotPresentInIngestedRundowns(ingestedRundowns)
+
+    await Promise.all(ingestedRundowns.map(async (ingestedRundown) => {
```

Comment:
The lambda function could be extracted to a new method e.g. `synchronizeIngestedRundown`.

Reply:
I considered it, but decided not because I don't think we gain that much from it.